logstreams/src/main/java/io/zeebe/logstreams/impl/backpressure/IncreasingClock.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.zeebe.logstreams.impl.backpressure;
+
+import java.util.function.Supplier;
+
+/**
+ * Supplies value of System.nanoTime() but ensures that values are always larger than previous ones.
+ * This ensure that measurements round-trip times in are always positive despite with potential
+ * concurrency or resolution issues in System.nanoTime() measurements.
+ */
+@SuppressWarnings("java:S4276")
+public class IncreasingClock implements Supplier<Long> {
+
+  private Long lastTime = null;
+
+  @Override
+  public synchronized Long get() {
```

Comment:
I'm not sure about adding another lock on every append/every incoming command, but I guess there's no way around it for now.

Reply:
AtomicLong compareAndSet?

Reply:
Similar to something like this https://github.com/camunda-cloud/zeebe/blob/develop/util/src/main/java/io/zeebe/util/sched/channel/ActorConditions.java#L33-L41 I see no need for a lock here

Reply:
Why is this necessary? The rate limiter listeners are created inside actors which eliminates contention in the first call and if there's contention is the second call it's probably very small. Even if there was some, I don't think this would be the bottleneck, if you think about the lifecycle of the request. Just my opinion though, I may be wrong